nav2_system_tests/src/system/tester_node.py

```diff
@@ -203,22 +203,13 @@ def shutdown(self):
         except Exception as e:
             self.error_msg('Service call failed %r' % (e,))
 
-
-def test_InitialPose(robot_tester, timeout, retries):
-    robot_tester.initial_pose_received = False
-    retry_count = 1
-    while not robot_tester.initial_pose_received and retry_count <= retries:
-        retry_count += 1
-        robot_tester.info_msg('Setting initial pose')
-        robot_tester.setInitialPose()
-        robot_tester.info_msg('Waiting for amcl_pose to be received')
-        rclpy.spin_once(robot_tester, timeout_sec=timeout)  # wait for poseCallback
-
-    if (robot_tester.initial_pose_received):
-        robot_tester.info_msg('test_InitialPose PASSED')
-    else:
-        robot_tester.info_msg('test_InitialPose FAILED')
-    return robot_tester.initial_pose_received
+    def wait_for_initial_pose(self):
```

Comment:
Part of the reasons for the timeouts is in case the test is flaky, that our tests don't take super long. That hasn't been a problem in recent months though so I think that can be removed.

Reply:
In `test_InitialPose()` routine `retry_count` guarantees that initial pose will set in a finite time interval. Otherwise, `test_InitialPose()` will return `False` value and further testing won't be continued.
Removing of `retry_count` may lead to go to infinite `while (not self.initial_pose_received) ...` loop and finally cause one more case to test timeout. Also, removing the return value of `test_InitialPose()` will break test code logic where depending on `result=test_InitialPose()` further testing will continue or not.

I am not sure, which problem this change is used to fix?

Reply:
OK, got it. That should work. Thank you for explanation!